apps/api/v2/src/modules/slots/slots-2024-09-04/controllers/e2e/reschedule-uid-slots.controller.e2e-spec.ts

```diff
@@ -0,0 +1,344 @@
+import { Test } from "@nestjs/testing";
+import { User, Booking } from "@prisma/client";
+import * as request from "supertest";
+import { BookingsRepositoryFixture } from "test/fixtures/repository/bookings.repository.fixture";
+import { EventTypesRepositoryFixture } from "test/fixtures/repository/event-types.repository.fixture";
+import { UserRepositoryFixture } from "test/fixtures/repository/users.repository.fixture";
+import { randomString } from "test/utils/randomString";
+
+import { CAL_API_VERSION_HEADER, SUCCESS_STATUS, VERSION_2024_09_04 } from "@calcom/platform-constants";
+import { CreateScheduleInput_2024_06_11 } from "@calcom/platform-types";
+
+describe("Slots 2024-09-04 Endpoints", () => {
+  describe("rescheduleUid functionality", () => {
+    let app: INestApplication;
+
+    let userRepositoryFixture: UserRepositoryFixture;
+    let schedulesService: SchedulesService_2024_06_11;
+    let eventTypesRepositoryFixture: EventTypesRepositoryFixture;
+    let bookingsRepositoryFixture: BookingsRepositoryFixture;
+
+    const userEmail = `slots-reschedule-uid-${randomString()}@example.com`;
+    let user: User;
+    let eventTypeId: number;
+    let eventTypeSlug: string;
+    let existingBooking: Booking;
+
+    beforeAll(async () => {
+      const moduleRef = await Test.createTestingModule({
+        imports: [
+          AppModule,
+          PrismaModule,
+          UsersModule,
+          TokensModule,
+          SchedulesModule_2024_06_11,
+          SlotsModule_2024_09_04,
+        ],
+      })
+        .overrideGuard(PermissionsGuard)
+        .useValue({
+          canActivate: () => true,
+        })
+        .compile();
+
+      userRepositoryFixture = new UserRepositoryFixture(moduleRef);
+      schedulesService = moduleRef.get<SchedulesService_2024_06_11>(SchedulesService_2024_06_11);
+      eventTypesRepositoryFixture = new EventTypesRepositoryFixture(moduleRef);
+      bookingsRepositoryFixture = new BookingsRepositoryFixture(moduleRef);
+
+      user = await userRepositoryFixture.create({
+        email: userEmail,
+        name: userEmail,
+        username: userEmail,
+      });
+
+      const userSchedule: CreateScheduleInput_2024_06_11 = {
+        name: `reschedule-uid-schedule-${randomString()}`,
+        timeZone: "Europe/Rome",
+        isDefault: true,
+      };
+      // note(Lauris): this creates default schedule monday to friday from 9AM to 5PM in Europe/Rome timezone
+      await schedulesService.createUserSchedule(user.id, userSchedule);
+
+      const event = await eventTypesRepositoryFixture.create(
+        {
+          title: "reschedule test meeting",
+          slug: `reschedule-uid-event-type-${randomString()}`,
+          length: 60,
+        },
+        user.id
+      );
+      eventTypeId = event.id;
+      eventTypeSlug = event.slug;
+
+      // Create an existing booking that will be used for reschedule testing
+      const bookedStartTime = "2050-09-05T11:00:00.000Z";
```

Comment:
We can put `const bookedStartTime` more up on the level where we define `let user: User;` so that tests below dont have to hardcode `2050-09-05T11:00:00.000Z` but instead can re-use the variable `bookedStartTime`.